sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/jdbc/JDBCTableCatalog.scala

```diff
@@ -282,12 +282,12 @@ class JDBCTableCatalog extends TableCatalog with SupportsNamespaces with Logging
       namespace: Array[String],
       cascade: Boolean): Boolean = namespace match {
     case Array(db) if namespaceExists(namespace) =>
-      if (listTables(Array(db)).nonEmpty) {
+      if (!cascade && listTables(Array(db)).nonEmpty) {
```

Comment:
`listTables` could be expensive. Can we try-catch the exception thrown by JDBC and detect the case when the schema/database is not empty?

Reply:
OK